autogen/agentchat/contrib/vectordb/base.py

```diff
@@ -0,0 +1,195 @@
+from typing import Any, Callable, List, Protocol, runtime_checkable, Dict
+
+
+@runtime_checkable
+class VectorDB(Protocol):
+    """
+    Abstract class for vector database. A vector database is responsible for storing and retrieving documents.
+    """
+
+    def __init__(self, db_config: Dict = None) -> None:
+        """
+        Initialize the vector database.
+
+        Args:
+            db_config: Dict | configuration for initializing the vector database. Default is None.
+
+        Returns:
+            None
+        """
+        ...
```

Comment:
Given this is so broad I'd say just don't include `__init__` in the protocol

Reply:
Right, the initializing  process vary from db to db.